dotnet/src/Connectors/Connectors.AI.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,209 +3,173 @@
+        var size = ImageSize.Size256x256;
+        switch (width)
         {
-            throw new KernelException("Azure OpenAI Text To Image result not found");
+            case 256:
+                size = ImageSize.Size256x256;
+                break;
+            case 512:
+                size = ImageSize.Size512x512;
+                break;
+            case 1024:
+                size = ImageSize.Size1024x1024;
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(width),
+                    "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024");
         }
 
-        return result.Result.Images.First().Url;
-    }
-
-    /// <summary>
-    /// Start an text to image task
-    /// </summary>
-    /// <param name="description">Image description</param>
-    /// <param name="width">Image width in pixels</param>
-    /// <param name="height">Image height in pixels</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns> The operationId that identifies the original text to image request. </returns>
-    private async Task<string> StartTextToImageAsync(string description, int width, int height, CancellationToken cancellationToken = default)
-    {
-        Verify.NotNull(description);
-        if (width != height || (width != 256 && width != 512 && width != 1024))
+        Response<ImageGenerations> imageGenerations;
+        try
         {
-            throw new ArgumentOutOfRangeException(nameof(width), width, "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024.");
+            imageGenerations = await this._client.GetImageGenerationsAsync(
+                new ImageGenerationOptions
+                {
+                    Prompt = description,
+                    Size = size,
+                }, cancellationToken).ConfigureAwait(false);
         }
-
-        var requestBody = JsonSerializer.Serialize(new TextToImageRequest
+        catch (RequestFailedException e) when (e.Status == 404)
         {
-            Prompt = description,
-            Size = $"{width}x{height}",
-            Count = 1
-        });
+            this._logger.LogError("Image generation failed with status code 404. This error can occur also when attempting to use Dall-E-3 which is still not supported");
+            throw;
+        }
 
-        var uri = this.GetUri(GenerationImageOperation);
-        var result = await this._core.ExecutePostRequestAsync<AzureOpenAITextToImageResponse>(uri, requestBody, cancellationToken).ConfigureAwait(false);
+        if (!imageGenerations.HasValue)
+        {
+            throw new KernelException("The response does not contain an image result");
+        }
 
-        if (result == null || string.IsNullOrWhiteSpace(result.Id))
+        if (imageGenerations.Value.Data.Count == 0)
```

Comment:
Can we add this condition to the previous one and have just:
```csharp
if (!imageGenerations.HasValue || imageGenerations.Value.Data.Count == 0)
{
    throw new KernelException("The response does not contain an image result");
}
```

Reply:
aren't they 2 different errors? one is about the result type, the other about the number of images? I'd prefer specific error messages to make debugging easier

Reply:
They are different, but from the user perspective, there is no difference between `The response does not contain an image result` exception message and `The response does not contain any image` exception message. I'm not sure if there is a case when user needs to handle these exceptions in different ways.